training/src/anemoi/training/commands/mlflow.py

```diff
@@ -129,7 +173,51 @@ def run(args: argparse.Namespace) -> None:
                 args.export_deleted_runs,
                 log_level,
             ).sync()
-            return
+            return None
+
+        if args.subcommand == "prepare":
+            # Load configuration
+            with initialize(version_base=None, config_path="./"):
+                cfg = compose(config_name=args.config_name)
+
+            # Create MLflow client and get experiment
+            client = AnemoiMlflowClient(cfg.diagnostics.log.mlflow.tracking_uri, authentication=True)
+            experiment_id = client.get_experiment_by_name(cfg.diagnostics.log.mlflow.experiment_name).experiment_id
+
+            # Parse configuration
+            if cfg.training.run_id is not None:  # Existing run_id
+                LOGGER.info("Existing run_id: %s", cfg.training.run_id)
+                try:
+                    client.get_run(cfg.training.run_id)
+                except ValueError as e:
+                    e(f"Run_id {cfg.training.run_id} not found. Modify your configuration file.")
+                    return None
+                return cfg.training.run_id
+
+            # Existing fork_id needs a new run_id attached to it
+            run = client.create_run(experiment_id, run_name=cfg.diagnostics.log.mlflow.run_name)
+            run_id = run.info.run_id
+            LOGGER.info("Creating new run_id: %s", run_id)
+
+            # Log the configuration file as an artifact
+            mlflow.set_tracking_uri(cfg.diagnostics.log.mlflow.tracking_uri)
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                fp = Path(tmp_dir, "config.json")
+                json.dump(OmegaConf.to_container(cfg), Path.open(fp, "w"))
+                client.log_artifact(run.info.run_id, fp)
+                client.set_tag(run_id, "mlflow.user", args.owner)
+                client.set_tag(run_id, "mlflow.source.name", "anemoi-training train")
```

Comment:
Should we set this tag to `anemoi-training prepare` , as a way to know which action created the tag? Then when training is started, the tag will be overwritten to `train` so you know training has run.